### PR TITLE
lib,commonlib: implement persistent logo

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

File: src/lib/coreboot_table.c [hunk 1]
```diff
@@ -408,6 +410,45 @@ void __weak lb_board(struct lb_header *header) { /* NOOP */ }
...
+	struct region_device rdev;
+	struct bootlogo_header *header = NULL;
+	size_t logo_size;
+
+	if (fmap_locate_area_as_rdev("BOOTSPLASH", &rdev))
+		return;
+
+
+	/* Make the buffer larger than the region. LZMA compressed BMP may be 4 times bigger. */
+	logo_size = (region_device_sz(&rdev) * 4);
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/lib/coreboot_table.c:428]: But someone may leave LINEAR_FRAMEBUFFER_MAX_WIDTH and LINEAR_FRAMEBUFFER_MAX_WIDTH as default values which will not work well for 4K monitors if a big logo would be added.

File: src/lib/coreboot_table.c [hunk 2]
```diff
@@ -408,6 +410,45 @@ void __weak lb_board(struct lb_header *header) { /* NOOP */ }
  */
 void __weak lb_spi_flash(struct lb_header *header) { /* NOOP */ }
 
+/*
```
Comment (mkopec) [inline src/lib/coreboot_table.c:413]: Have you seen clevo/release implementation for the same thing? It was updated for the newer CBFS API and avoids manual buffer allocation.

https://github.com/Dasharo/coreboot/blob/ae10b20f5c6abc9c23f709b65c46be6525da8c13/src/lib/coreboot_table.c#L408-L434

Comment (miczyg1) [reply src/lib/coreboot_table.c:413]: I can rewrite it to such state.

File: src/lib/coreboot_table.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/lib/coreboot_table.c]: After addressing this discussion https://github.com/Dasharo/coreboot/pull/352#discussion_r1250788332
The comment is no longer relevant

File: src/lib/coreboot_table.c (hunk 2 above)
Comment (miczyg1) [reply src/lib/coreboot_table.c:414]: @mkopec done

Comment (miczyg1) [reply src/lib/coreboot_table.c:413]: At this point it made more sense to simply cherry-pick your commit.

Comment (mkopec) [reply src/lib/coreboot_table.c:413]: Thanks

**Review by mkopec — APPROVED** [review]

**Pull request merged**